collections/duties.js

```diff
@@ -0,0 +1,33 @@
+function newDutie(data) {
+  const collection = require("../start").dutiesCollection;
+  const resHandler = require("../start").server.getResHandler();
+  const fields = ["name", "location", "days", "constraints", "soldiersRequired", "value"];
+
+  for (let i = 0; i < fields.length; i++) {
+    if (!data[fields[i]]) {
+      resHandler.notValidData("The dutie json that was sent is missing the field " + fields[i]);
+      return;
+    }
+  }
+
+  if (Object.keys(data).length != 6) {
```

Comment:
why the magic number? fields.length and it's clearer by a margin. NEVER USE MAGIC NUMBERS MAGIC ISN'T REAL